wys/api/sql/create-view-mobile_api_id.sql

```diff
@@ -2,27 +2,59 @@ DROP VIEW wys.mobile_api_id CASCADE;
 
 CREATE MATERIALIZED VIEW wys.mobile_api_id AS
 
-SELECT id AS location_id,
-a.ward_no, a.location, a.from_street, a.to_street, a.direction, 
-a.installation_date, a.removal_date, a.comments, a.combined,
-b.api_id
-FROM (
-SELECT *, 
- CASE WHEN new_sign_number NOT LIKE 'W%' THEN 'Ward ' || ward_no || ' - S' || new_sign_number 
- WHEN new_sign_number LIKE 'W%' AND new_sign_number LIKE '% - S%' THEN 'Ward ' || SUBSTRING(new_sign_number, 2, 10)
- WHEN new_sign_number LIKE 'W%' AND new_sign_number NOT LIKE '% - S%' THEN 'Ward ' || SUBSTRING(new_sign_number, 2, POSITION(' - ' IN new_sign_number) + 1 ) || 'S' || RIGHT(new_sign_number, 1)
- END AS combined
-FROM wys.mobile_sign_installations
-) a 
-LEFT JOIN 
-(SELECT DISTINCT api_id, sign_name, start_date,
-                 lead(start_date) OVER w AS next_start,
-                 lag(start_date) OVER w AS prev_start
-FROM wys.locations
-WHERE sign_name LIKE 'Ward%'
-WINDOW w AS (PARTITION BY sign_name ORDER BY start_date)) b
-ON a.combined = b.sign_name
-   AND (b.prev_start IS NULL OR installation_date >= b.prev_start)
-   AND (b.next_start IS NULL OR installation_date < b.start_date);
+WITH sign_locations AS (
+    SELECT DISTINCT 
+        api_id,
+        sign_name,
+        start_date,
+        lead(start_date) OVER w AS next_start,
+        lag(start_date) OVER w AS prev_start
+    FROM wys.locations
+    WHERE sign_name LIKE 'Ward %'
+    WINDOW w AS (PARTITION BY sign_name ORDER BY start_date)
+),
 
-CREATE UNIQUE INDEX ON  wys.mobile_api_id (location_id);
\ No newline at end of file
+mobile_installations AS (
+    SELECT
+        ward_no,
+        location,
+        from_street,
+        to_street,
+        direction,
+        installation_date,
+        removal_date,
+        new_sign_number,
+        comments,
+        id, 
+        CASE WHEN new_sign_number NOT LIKE 'W%'
+            THEN 'Ward ' || ward_no || ' - S' || new_sign_number
+            WHEN new_sign_number LIKE 'W%' AND new_sign_number LIKE '% - S%'
+                THEN 'Ward ' || SUBSTRING(new_sign_number, 2, 10)
+            WHEN new_sign_number LIKE 'W%' AND new_sign_number NOT LIKE '% - S%'
+                THEN 'Ward '
+                || SUBSTRING(new_sign_number, 2, POSITION(' - ' IN new_sign_number) + 1)
+                || 'S' || RIGHT(new_sign_number, 1)
+        END AS combined
+    FROM wys.mobile_sign_installations
+)
+
+
+SELECT 
+    msi.id AS location_id,
+    msi.ward_no,
+    msi.location,
+    msi.from_street,
+    msi.to_street,
+    msi.direction,
+    msi.installation_date,
+    msi.removal_date,
+    msi.comments,
+    msi.combined,
+    b.api_id
+FROM mobile_installations AS msi
+LEFT JOIN sign_locations AS b
+    ON msi.combined = b.sign_name
+    AND (b.prev_start IS NULL OR msi.installation_date >= b.prev_start)
+    AND (b.next_start IS NULL OR msi.installation_date < b.start_date);
```

Comment:
@Nate-Wessel Do we really need double tabs here?

Reply:
That's what SQLFLuff suggests for multiple join conditions. I didn't really like it at first, but to be honest it's kind of growing on me. My personal inclination would be to do something like 
```sql
JOIN a ON
    x = y
    AND b = c
```
🤷